io.github.kalaksi.Lightkeeper.yml

```diff
@@ -0,0 +1,58 @@
+app-id: io.github.kalaksi.Lightkeeper
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+sdk-extensions:
+  - "org.freedesktop.Sdk.Extension.rust-stable"
+command: lightkeeper
+finish-args:
+  - "--share=ipc"
+  - "--socket=fallback-x11"
+  - "--socket=wayland"
+  - "--share=network"
+  - "--device=dri"
+  - "--socket=ssh-auth"
+build-options:
+    append-path: /usr/lib/sdk/rust-stable/bin
+    build-args:
+      - "--share=network"
+    env:
+      RUSTFLAGS: "--remap-path-prefix =../"
+      CARGO_HOME: /run/build/lightkeeper/cargo
+modules:
+  # Liboping is needed for ping monitoring module.
+  - name: liboping
+    buildsystem: simple
+    sources:
+      - type: archive
+        url: https://noping.cc/files/liboping-1.10.0.tar.gz
+        sha256: c206b05743d0730814be3115b48abd0b00016677525153c78730da307aba0846
+      # Apparently liboping isn't well maintained upstream, so we need patches from Debian.
+      - type: archive
+        url: https://deb.debian.org/debian/pool/main/libo/liboping/liboping_1.10.0-5.debian.tar.xz
+        sha256: bd2d6bd63d032fb4d5a65ed74467baffea74e902d13fdd8409fba3ede35978a4
+        dest: debian-package
+    build-commands:
+      # Remove one patch that modifies configure.ac to avoid needing aclocal-1.14.
+      - rm debian-package/patches/0002-autoupdate-and-enable-automake-Wall.patch
+      - for patchfile in debian-package/patches/*.patch; do patch -p1 < "$patchfile"; done
+      - ./configure --without-perl-bindings --prefix=/app
```

Comment:
this should be a `buildsytem: autotools` build. you can add a shell to do some pre-processing.